### PR TITLE
Updated docs to follow our standard conf

### PR DESCRIPTION
**Pull request by philipstarkey**
Also added links to other labscript suite docs and standard links.

### CODE REVIEW

**Comment by philipstarkey** [comment]
Note that this is currently not building on RTD: https://readthedocs.org/projects/philipstarkey-labscript/builds/11263059/

RTD is failing to install the labscript_utils dependency - not sure if that is expected or not given the current tagged release state...

**Comment by rpanderson** [comment]
> RTD is failing to install the labscript_utils dependency - not sure if that is expected or not given the current tagged release state...

It might be because there are only pre-releases of labscript-utils and other components. I can reproduce the failure locally in a fresh venv and clone of labscript.
```
(.venv) > python -m pip install --upgrade --upgrade-strategy eager --no-cache-dir .[docs]
ERROR: Could not find a version that satisfies the requirement labscript-utils>=2.14.0 (from labscript==3.0.0rc2.dev2+g28dc675) (from versions: 2.15.1.dev54, 3.0.0b2, 3.0.0rc1)
ERROR: No matching distribution found for labscript-utils>=2.14.0 (from labscript==3.0.0rc2.dev2+g28dc675)
```
Adding `--pre` resolves the issue, but this cannot be specified on RTD (see readthedocs/readthedocs.org#5018). 

This error does arise when following either our regular or developer installation guide for pip.

**Comment by rpanderson** [comment]
One workaround (until we tag fully qualified releases) is to [include a pre-release or development version](https://pip.pypa.io/en/1.4.1/logic.html?highlight=requirements#pre-release-versions) in the offending `install_requires` entry in setup.cfg, e.g.
```ini
[options]
...
install_requires =
  ...
  labscript_utils>=2.14.0dev0
```
Working build with this modification [here](https://rpanderson-labscript.readthedocs.io/) ([build log](https://readthedocs.org/projects/rpanderson-labscript/builds/11270347/)).

Alternatively, so that pre-releases are only allowed for documentation builds, we could include docs/requirements.txt the above requirement specification and add the following to readthedocs.yaml:
```yaml
python:
  install:
    - requirements: docs/requirements.txt
    ...
```

**Comment by philipstarkey** [comment]
Cool! I've updated the requirement to >=3.0.0rc1. We needed to bump it to 3.0.0 anyway, so I don't think there is any downside to doing this (even if we forget to remove it - which hopefully we won't!)

Changes viewable here: https://philipstarkey-labscript.readthedocs.io/en/latest/components/

File: docs/source/index.rst [hunk 1]
```diff
@@ -5,23 +5,33 @@
 
 ``labscript`` documentation
```
Comment (rpanderson) [inline docs/source/index.rst:6]: Alternatively, `labscript Documentation` (regular font style for project and/or capitalise Documentation)?

Comment (philipstarkey) [reply docs/source/index.rst]: Yep. This was left over from the original Sphinx docs I started circa 2015. It's now updated to match the other docs.

**Pull request merged**